dicomImport/matRad_scanDicomImportFolder.m

```diff
@@ -46,11 +48,35 @@
 fileList = {mainDirInfo(~dirIndex).name}';
 patientList = 0;
 
+% add files of subfolders
+% for i = 1:3
+subFolderList = {mainDirInfo(dirIndex).name}';
+subFolderList(ismember(subFolderList,{'.','..'})) = [];
+subFolderList = cellfun(@(x) fullfile(patDir,x),...
+        subFolderList, 'UniformOutput', false);
+subFolderFileList = [];
+for i = 1 : numel(subFolderList)
+    % loop subfolders
+    subDir = subFolderList{i};
+    % get information about main directory
+    subDirInfo = dir(subDir);
+    % get index of subfolders
+    dirIndex = [subDirInfo.isdir];
+    % list of filenames in main directory
+    subFileList = {subDirInfo(~dirIndex).name}';
+    if ~isempty(subFileList)
+        subFileList = cellfun(@(x) fullfile(subFolderList{i},x),...
+            subFileList, 'UniformOutput', false);
+    end
+    subFolderFileList = [subFolderFileList; subFileList];
+end
+
 % create full path for all files in main directory
 if ~isempty(fileList)
     fileList = cellfun(@(x) fullfile(patDir,x),...
         fileList, 'UniformOutput', false);
-    
+    % attach subfolderfiles to fileList
+    fileList = [fileList; subFolderFileList];
```

Comment:
I like the idea to also include subfolders but I do not understand why you add these to 'fileList' right here.... Is it possible that fileList is empty (because there are not files only folders and '..' and '.' in there)?